docs/docs/running-hello-world.md

```diff
@@ -73,265 +85,289 @@ The web server loads the enclave and waits for requests from the client. Amongst
-7. The host passed the encrypted message to the enclave.
-8. The enclave decrypted the client's message, reversed the string, and encrypted the response using the client's
-   public key.
-9. The enclave passed the encrypted message to the host, was sent back to the client as an HTTP response to the same
-   request.
 
-Congratulations on running your first enclave!
-
-### Enclave Constraint
-The `--constraint` parameter of the client command-line interface defines the 
-properties of an acceptable enclave. This includes the signing key 
-```
-S:0000000000000000000000000000000000000000000000000000000000000000
-```
-which we saw in the host output in the previous
-section.
+You have run your first enclave successfully.
 
-Try rerunning the command with a different code signer:
-```
-"S:2222222222222222222222222222222222222222222222222222222222222222 PROD:1 SEC:INSECURE"
-```
 
-You should see the following error:
-> com.r3.conclave.common.InvalidEnclaveException: Enclave code signer does not match any of the acceptable code signers.
-(key hash 0000000000000000000000000000000000000000000000000000000000000000 vs acceptable
-2222222222222222222222222222222222222222222222222222222222222222)
+## Sequence of events
 
-For more information on how to choose a constraint, see [Enclave constraints](constraints.md).
+The sample application reversed the string securely and confidentially by these steps:
 
+1. The host loads the enclave, which generates a remote attestation report.
+2. The client connects to the host and retrieves the attestation report.
+3. The client uses the attestation report to verify that it is connected to the expected enclave.
+4. The client encrypts the "reverse-me" string using the public key provided in the attestation.
+5. In the message to the enclave, the client also includes a public key for the enclave to use when responding.
+6. The client sends the encrypted message to the host's REST API.
+7. The host passes the encrypted message to the enclave.
+8. The enclave decrypts the client's message, reverses the string, and encrypts the response using the client's
+   public key.
+9. The enclave passes the encrypted response to the host, which forwards it to the client in an HTTP response.
 
 ## Beyond mock mode
-Eventually you will want to run your application in one of the other [enclave modes](enclave-modes.md), namely simulation,
-debug, or release. This section describes how to build and run the sample in these other modes.
 
-**First, make sure that your system fulfills the
-[system requirements](enclave-modes.md#system-requirements) for your desired mode.**
+In the previous section, you run the sample application in [mock mode](enclave-modes.md). 
+This section describes how to build and run the sample in other [enclave modes](enclave-modes.md), namely
+simulation, debug, and release.
+
+*First, ensure that your system fulfills the [system requirements](enclave-modes.md#system-requirements) for
+your desired mode.*
```

Comment:
Not "desired mode", which sort of implies that debug/simulation are happy compromises to release. We need to quickly, but firmly, state here that only release mode gives you confidential guarantees of SGX.